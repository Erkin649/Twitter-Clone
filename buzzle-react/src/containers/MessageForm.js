import { connect } from 'react-redux';

import { clearError } from '../actions/error.js';
import { createMessage } from '../actions/messages.js';
import {
    updateCreateMessageForm,
    clearCreateMessageForm,
    unloadCreateMessageForm
} from '../actions/forms.js';
import MessageForm from '../components/MessageForm.js';

const mapStateToProps = state => ({
    'user': state.user,
    'content': state.createMessageForm.content
});

const mapDispatchToProps = dispatch => ({
    'clearError': () => dispatch(clearError()),
    'createMessage': (content, onSuccess) => dispatch(createMessage(content, onSuccess)),
    'updateCreateMessageForm': (name, value) => dispatch(updateCreateMessageForm(name, value)),
    'clearCreateMessageForm': () => dispatch(clearCreateMessageForm()),
    'unloadCreateMessageForm': () => dispatch(unloadCreateMessageForm())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(MessageForm);
