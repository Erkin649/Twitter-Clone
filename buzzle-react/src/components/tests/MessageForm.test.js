import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import MessageForm from '../MessageForm.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageForm />, div);
});

it('renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<MessageForm />)
                         .toJSON();

    expect(tree).toMatchSnapshot();
});