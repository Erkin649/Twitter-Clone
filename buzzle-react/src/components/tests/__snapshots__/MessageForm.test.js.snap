// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and equal to snapshot 1`] = `null`;
